gammapy/datasets/spectrum.py

```diff
@@ -325,15 +327,28 @@ def read(cls, filename, format="ogip", checksum=False, name=None, **kwargs):
         """
         from .io import OGIPDatasetReader
 
-        if format == "gadf":
-            return super().read(
-                filename, format="gadf", checksum=checksum, name=name, **kwargs
-            )
+        if format is None:
+            with fits.open(filename) as hdulist:
+                # Check if the file is empty
+                if len(hdulist) == 0:
+                    raise ValueError(f"File {filename} is empty")
+                # Check for extensions in OGIP format
+                if "SPECTRUM" in hdulist and "OGIP" in hdulist[1].header["HDUCLASS"]:
+                    format = "ogip"
+                # Check for extensions in GADF format
+                elif ("COUNTS" in hdulist and "EXPOSURE" in hdulist and "EDISP" in hdulist):
+                    format = "gadf"
+                    if 'IMAGE' in hdulist[1].header["XTENSION"]:
+                        raise ValueError(f"File {filename} is not a GADF spectrum, but a GADF map")
```

Comment:
Maybe check that the `REGION` HDU is there instead?

Reply:
if REGION is there it should be a SpectrumDatasetOnOff right? If so, this is implemented now.